### ISSUE TITLE
How to use SPK_CONFLICT

### ISSUE THREAD

**Issue by agross** [SynoCommunity/spksrc#2893]
### Setup

I try to use `SPK_CONFLICT` (which is undocumented, BTW) for a new package that conflicts with say, the `foo` package.

As soon as I add 

```Makefile
SPK_CONFLICT = foo
```

to my `spk/test/Makefile` (copied from `spk/zsh`) the builds fails:

```
$ cd /sprksrc/spk/test
$ make
../../mk/spksrc.spk.mk:146: *** recipe commences before first target.  Stop.
```

I'm using docker to build.

**Comment by m4tt075** [comment]
I have no experience with SPK_CONFLICT, but used a PKG_CONX file to avoid conflicts with another package (see PR #2801 for details). Might be an alternative to achieve what you are looking for...

**Comment by agross** [comment]
Hi @m4tt075,

Thanks for the hint. How is the `PKG_CONX` file evaluated, or rather who evaluates it? `grep -r PKG_CONX .` shows no results for me.

Thanks!

**Comment by m4tt075** [comment]
The `PKG_CONX` file is not processed by the cross-compilation framework (`spksrc`) but by DSM on installation. It is part of the spk specifications (see Synology DSM6 Development Guide).

**Comment by cytec** [comment]
@agross i've not used it before but i guess it'll work like `SPK_DEPENDS`.. so try something like this: `SPK_CONFLICT="foo>1.2"`

**Comment by ymartin59** [comment]
`SPK_CONFLICT` is simply copied as `install_conflict_packages` in `INFO` file. Refer to https://developer.synology.com/download/developer-guide.pdf
According to that documentation, `PKG_CONX` has higher priority over `INFO` fields.

You have to create `src/conf/PKG_CONX` file in your `spk` with content like examples shown page 70.

By the way I do not understand yet why `install_conflict_packages` from `INFO` may not be honored.

**Comment by agross** [comment]
Thanks for the explanation and details.

My current problem with SPK_CONFLICT is not an issue with the info not being honored, it is that the build fails very early with the message in the original post.

**Comment by cytec** [comment]
@agross have you tried changing 

```
SPK_CONFLICT = foo
```
to
```
SPK_CONFLICT="foo"
```

**Comment by agross** [comment]
Yes, I did. It all leads to the same error.

Did you try that syntax and did it work for you?

**Comment by agross** [comment]
Tested a bit more. GNU make 4.0 (docker) and 4.2.1 (on my Mac) both fail with the same error as soon as I `@echo anything` inside `ifneq ($(strip $(SPK_CONFLICT)),)`.

**Comment by m4tt075** [comment]
You don't need to bother, as looking at your PR you have everything in place. You can still share the cross package as you do right now:
1. Ensure the folder `spk/zsh/src/conf/` exists. If not, create it.
2. Create a `PKG_CONX` file there with content `[zsh-static]`. Nothing else.
3. Ensure the folder `spk/zsh-static/src/conf/` exists. If not, create it.
4. Create a `PKG_CONX` file there with content `[zsh]`. 
5. Make clean and rebuild both packages and you should be done.

**Comment by ymartin59** [comment]
I investigated `spksrc.spk.mk` by the way... it is really strange `SPK_CONFLICT` get `make` crash... Troubles come from previous switch for `DEBUG` flag.
If I comment the `DEBUG` section or if I move `SPK_CONFLICT` session before `DEBUG`, then `make` is just happy

**Comment by ymartin59** [comment]
Here is the fix #2903. If `SPK_CONFLICT` is enough for a conflict declaration, I would prefer its usage instead of providing "complex" `PKG_CONX` file.
